Build using musl (on Alpine Linux) fails

## Summary

I'm currently trying to compile ClangBuildAnalyzer on Alpine Linux, which uses musl and not glibc. That fails due to `ftello64` being a glibc-only extension (a plain `ftello` is apparently sufficient for musl, see [here](https://gitlab.alpinelinux.org/alpine/aports/-/issues/16851#note_474711) for example).

## Reproducer

Inside an Alpine container that has access to the sources (i.e. `docker run -it --volume [...] alpine:3.21`):

```bash
$ apk add gcc g++ cmake ninja
$ cd /tmp
$ mkdir build && cd build
$ cmake -GNinja -DCMAKE_BUILD_TYPE=Release /src/ClangBuildAnalyzer
$ ninja
```

Results in those compiler errors:

```
ninja: job failed: /usr/bin/c++   -O3 -DNDEBUG -MD -MT CMakeFiles/ClangBuildAnalyzer.dir/src/main.cpp.o -MF CMakeFiles/ClangBuildAnalyzer.dir/src/main.cpp.o.d -o CMakeFiles/ClangBuildAnalyzer.dir/src/main.cpp.o -c /ext/clang-build-analyzer/ClangBuildAnalyzer-1.6.0/src/main.cpp
/ext/clang-build-analyzer/ClangBuildAnalyzer-1.6.0/src/main.cpp: In function 'void ReadFileToString(const std::string&, std::string&)':
/ext/clang-build-analyzer/ClangBuildAnalyzer-1.6.0/src/main.cpp:39:20: error: 'ftello64' was not declared in this scope; did you mean 'ftello'?
   39 |     size_t fsize = ftello64(f);
      |                    ^~~~~~~~
      |                    ftello
ninja: job failed: /usr/bin/c++   -O3 -DNDEBUG -MD -MT CMakeFiles/ClangBuildAnalyzer.dir/src/BuildEvents.cpp.o -MF CMakeFiles/ClangBuildAnalyzer.dir/src/BuildEvents.cpp.o.d -o CMakeFiles/ClangBuildAnalyzer.dir/src/BuildEvents.cpp.o -c /ext/clang-build-analyzer/ClangBuildAnalyzer-1.6.0/src/BuildEvents.cpp
/ext/clang-build-analyzer/ClangBuildAnalyzer-1.6.0/src/BuildEvents.cpp: In constructor 'BufferedReader::BufferedReader(FILE*)':
/ext/clang-build-analyzer/ClangBuildAnalyzer-1.6.0/src/BuildEvents.cpp:558:24: error: 'ftello64' was not declared in this scope; did you mean 'ftello'?
  558 |         size_t fsize = ftello64(f);
      |                        ^~~~~~~~
      |                        ftello
ninja: subcommands failed
```

I have a preliminary fix for this at #104 .

--- a/src/BuildEvents.h
+++ b/src/BuildEvents.h
@@ -11,7 +11,7 @@
 
 #ifdef _MSC_VER
 #define ftello64 _ftelli64
-#elif defined(__APPLE__) || defined(__FreeBSD__)
+#elif defined(__APPLE__) || defined(__FreeBSD__) || (defined(__linux__) && !defined(__GLIBC__))
 #define ftello64 ftello
 #endif
 

